runtime/Go/antlr/tokenstream_rewriter_test.go

```diff
@@ -10,6 +10,16 @@ import (
 	"strings"
 )
 
+
+/* Assume the following grammar for this test.
+
+lexer grammar LexerA;
+A : 'a';
+B : 'b';
+C : 'c';
+
+*/
```

Comment:
Where did you take it from? Probably it should be extracted to a separate file.

Reply:
Of course it should be in a file. But, so should the grammar in the comment [here](https://github.com/antlr/antlr4/blob/f5f777d5af3a8e847465787e6112872065440014/runtime/Go/antlr/testing_lexer_b_test.go#L12).

But, even if we decide to leave it here, and pull out the grammars into files, and generate the lexers for testing, the generated code will not build. That's because we have the lovely [global variable problem](https://github.com/antlr/antlr4/issues/3452). And, rather than deal with the problem in an intelligent manner, the output from the tool is "cleverly" hand-edited.

Even if we fix #3452, API tests should not even go here. How can one test the visibility of the API [from within the package[(https://github.com/antlr/antlr4/blob/f5f777d5af3a8e847465787e6112872065440014/runtime/Go/antlr/testing_lexer_b_test.go#L5)? It's all just wrong. The tests should not be here. It needs to go into the [runtime-testsuite](https://github.com/antlr/antlr4/tree/dev/runtime-testsuite) directory.

Reply:
> So, who knows if this is right. The "experts" are so much smarter than I. I only have a ~1/2 century of programming that I have done, but it's nothing.

Everyone can mistake despite the big experience as well as suggest a good solution. And it's okay.

Reply:
It's normal to make mistakes. We are all human. It's wrong to not take ownership.  This has been here for a long time, enough time to reflect and try to fix.

I think it best to add the code as is. #3452 / #3264 needs to be addressed post 4.10. I have a fix for this, but it needs to be integrated with a fix with #3446, which I also have.